core/state_processor.go

```diff
@@ -121,6 +121,9 @@ func ApplyTransaction(config *params.ChainConfig, bc ChainContext, author *commo
 	// Set the receipt logs and create a bloom for filtering
 	receipt.Logs = statedb.GetLogs(tx.Hash())
 	receipt.Bloom = types.CreateBloom(types.Receipts{receipt})
+	receipt.BlockHash = header.Hash()
+	receipt.BlockNumber = header.Number
+	receipt.TransactionIndex = uint(statedb.TxIndex())
```

Comment:
A better solution might be to set them in Process, since you can reuse the hash instead of recalculating the header hash for every tx. 

The drawback, I guess, is that direct usage of `ApplytTransaction` won't get it, which I guess applies to the miner, the txpool (?) and various users of chain_makers in tests.

Reply:
@holiman fixed it now by adding an accessor for block hash to StateDB. The PR already adds an accessor for TxIndex, one more can't hurt.